alibuild_helpers/utilities.py

```diff
@@ -310,7 +328,12 @@ def getPackageList(packages, specs, configDir, preferSystem, noSystem,
   packages = packages[:]
   validDefaults = []  # empty list: all OK; None: no valid default; non-empty list: list of valid ones
   while packages:
-    p = packages.pop(0)
+    fullPkg = packages.pop(0)
+    target = all
+    if ":" in fullPkg:
+      p, target = fullPkg.split(":")
+    else:
+      p = fullPkg
     if p in specs:
       continue
```

Comment:
Should we really `continue` here? What if we need to build packages with conflicting targets?

For example, what should happen with an invocation such as `aliBuild build O2:analysis arrow:some-other-target`, where `O2:analysis` depends on `arrow`?

Should we propagate targets to dependencies, and build them twice if necessary? How do we keep them apart? (A lot of code currently assumes that if `spec1["package"] == spec2["package"]`, then `spec1 == spec2` -- which this would break.)